web/webpack.plugin.js

```diff
@@ -0,0 +1,207 @@
+const path = require('path');
+const DynamicRemotePlugin = require('@openshift/dynamic-plugin-sdk-webpack').DynamicRemotePlugin;
+const CSSMinimizerPlugin = require('css-minimizer-webpack-plugin');
+const webpack = require('webpack');
+const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
+
+const BG_IMAGES_DIRNAME = 'assets';
+
+const isProd = process.env.NODE_ENV === 'production';
+
+const pathTo = (relativePath) => path.resolve(__dirname, relativePath);
+
+/**
+ * Shared modules consumed and/or provided by this plugin.
+ *
+ * A host application typically provides some modules to its plugins. If an application
+ * provided module is configured as an eager singleton, we suggest using `import: false`
+ * to avoid bundling a fallback version of the module when building your plugin.
+ *
+ * Plugins may provide additional shared modules that can be consumed by other plugins.
+ *
+ * @see https://webpack.js.org/plugins/module-federation-plugin/#sharing-hints
+ */
+const pluginSharedModules = {
+  '@openshift/dynamic-plugin-sdk': { singleton: true, import: false },
+  '@patternfly/react-core': {},
+  '@patternfly/react-table': {},
+  react: { singleton: true, import: false },
+  'react-dom': { singleton: true, import: false },
+};
+
+const plugins = [
+  new webpack.EnvironmentPlugin({
+    NODE_ENV: 'development',
+  }),
+  new DynamicRemotePlugin({
+    sharedModules: pluginSharedModules,
+    entryScriptFilename: isProd ? 'plugin-entry.[fullhash].min.js' : 'plugin-entry.js',
+  }),
+];
+
+module.exports = {
+  mode: isProd ? 'production' : 'development',
+  entry: {}, // Plugin container entry is generated by DynamicRemotePlugin
+  output: {
+    path: pathTo('dist'),
+    publicPath: 'http://localhost:9001/',
```

Comment:
Something for the future, the public path needs to be adjusted to match the actual deployment once we get there.